src/main/scala/tech/cryptonomic/conseil/tezos/DatabaseConversions.scala

```diff
@@ -890,23 +893,26 @@ object DatabaseConversions extends LazyLogging {
 
     }
 
-  implicit val delegateToRow = new Conversion[Id, (BlockHash, Int, PublicKeyHash, Delegate), Tables.DelegatesRow] {
-    override def convert(from: (BlockHash, Int, PublicKeyHash, Delegate)) = {
-      val (blockHash, blockLevel, keyHash, delegate) = from
-      Tables.DelegatesRow(
-        pkh = keyHash.value,
-        blockId = blockHash.value,
-        balance = extractBigDecimal(delegate.balance),
-        frozenBalance = extractBigDecimal(delegate.frozen_balance),
-        stakingBalance = extractBigDecimal(delegate.staking_balance),
-        delegatedBalance = extractBigDecimal(delegate.delegated_balance),
-        rolls = delegate.rolls.getOrElse(0),
-        deactivated = delegate.deactivated,
-        gracePeriod = delegate.grace_period,
-        blockLevel = blockLevel
-      )
+  implicit val delegateToRow =
+    new Conversion[Id, (BlockHash, Int, PublicKeyHash, Delegate, Option[Int], Option[Int]), Tables.BakersRow] {
```

Comment:
Thoughts: Maybe we should start assigning a type to these bunches of values?
We can probably create Value Classes if the tuple is simply wrapped, with the inconvenience of lacking named fields.

Reply:
I think that could be a good idea, but as a part of a bigger refactor.